### ISSUE TITLE
Build failure: clamav-freshclam.service

### ISSUE THREAD

**Issue by BirdeeHub** [NixOS/nixpkgs#282450]
### Steps To Reproduce

in system nixos module, installed via flake nixos-rebuild switch
```nix
  services.clamav.daemon.enable = true;
  services.clamav.updater.enable = true;
  services.clamav.updater.interval = "weekly";
```
on nixpkgs version
```nix
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
```
```
• Updated input 'nixpkgs':
    'github:nixos/nixpkgs/842d9d80cfd4560648c785f8a4e6f3b096790e19' (2024-01-17)
  → 'github:nixos/nixpkgs/bbe7d8f876fbbe7c959c90ba2ae2852220573261' (2024-01-19)

```

### Build log

```
evaluating derivation 'git+file:///home/birdee/Projects/birdeeSystems#nevaluating derivation 'git+file:///home/birdee/Projects/birdeeSystems#nixosConfigurationerror:
       … while calling anonymous lambda

         at /nix/store/kwd6lmx004rkv2r00vj3fcg5ijfvnagk-source/lib/attrsets.nix:956:24:

          955|     let f = attrPath:
          956|       zipAttrsWith (n: values:
             |                        ^
          957|         let here = attrPath ++ [n]; in

       … while calling 'g'

...etc...

      … while evaluating the option `system.build.toplevel':

       … while calling anonymous lambda

         at /nix/store/kwd6lmx004rkv2r00vj3fcg5ijfvnagk-source/lib/modules.nix:824:28:

          823|         # Process mkMerge and mkIf properties.
          824|         defs' = concatMap (m:
             |                            ^
          825|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))

       … while evaluating definitions from `/nix/store/kwd6lmx004rkv2r00vj3fcg5ijfvnagk-source/nixos/modules/system/activation/top-level.nix':

       … from call site

         at /nix/store/kwd6lmx004rkv2r00vj3fcg5ijfvnagk-source/lib/modules.nix:825:137:

          824|         defs' = concatMap (m:
          825|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))
             |                                                                                                                                         ^
          826|         ) defs;

       … while calling 'dischargeProperties'

         at /nix/store/kwd6lmx004rkv2r00vj3fcg5ijfvnagk-source/lib/modules.nix:896:25:

          895|   */
          896|   dischargeProperties = def:
             |                         ^
          897|     if def._type or "" == "merge" then

       … from call site

         at /nix/store/kwd6lmx004rkv2r00vj3fcg5ijfvnagk-source/nixos/modules/system/activation/top-level.nix:71:12:
                    70|   # Replace runtime dependencies
           71|   system = foldr ({ oldDependency, newDependency }: drv:
             |            ^
           72|       pkgs.replaceDependency { inherit oldDependency newDependency drv; }

       … while calling 'foldr'

...etc...

       error:
       Failed assertions:
       - clamav-freshclam.service is ordered after 'network-online.target' but doesn't depend on it
```

### Additional context
I excerpted the interesting portions of the stack trace for you above.

Basically, I flake updated and clamav enabled via the 3 lines at the top failed to build.

### Notify maintainers

@robberer @qknight  @globin 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.73, NixOS, 24.05 (Uakari), 24.05.20240119.bbe7d8f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by e-nikolov** [comment]
I have a similar error but for [golink ](https://github.com/tailscale/golink) which comes from a separate flake not in nixpkgs

```
error:
       … while checking flake output 'nixosConfigurations'

       … while checking the NixOS configuration 'nixosConfigurations.home-nix'

       … while calling the 'head' builtin

         at /nix/store/kwd6lmx004rkv2r00vj3fcg5ijfvnagk-source/lib/attrsets.nix:960:11:

          959|         || pred here (elemAt values 1) (head values) then
          960|           head values
             |           ^
          961|         else

       (stack trace truncated; use '--show-trace' to show the full trace)

       error:
       Failed assertions:
       - golink.service is ordered after 'network-online.target' but doesn't depend on it
Error: Process completed with exit code 1.
```

**Comment by e-nikolov** [comment]
Looks like this is the relevant commit: https://github.com/NixOS/nixpkgs/commit/62f30634db2146c23f9aff1c9b598524652bbeb7

> `multi-user.target` no longer depends on `network-online.target`.
  This will potentially break services that assumed this was the case in the past.
  This was changed for consistency with other distributions as well as improved boot times.
> We have added a warning for services that are
  `after = [ "network-online.target" ]` but do not depend on it (e.g. using `wants`).

**Comment by BirdeeHub** [comment]
That is some good sleuthing. Seems likely

**Comment by e-nikolov** [comment]
For your case the solution is probably to add `wants = [ "network-online.target" ]` to the clamAV service which may have been missed by that PR.

**Comment by BirdeeHub** [comment]
Oh!! It very much was!!

**Comment by lf-** [comment]
Sneakily fixed by #282639
